contracts/fio.address/fio.address.cpp

```diff
@@ -523,26 +525,36 @@ namespace fioio {
                 chaincode = tpa->chain_code.c_str();
 
-                fio_400_assert(validateChainNameFormat(token), "token_code", tpa->token_code, "Invalid token code format",
+                fio_400_assert(validateTokenNameFormat(token), "token_code", tpa->token_code, "Invalid token code format",
                                ErrorInvalidFioNameFormat);
                 fio_400_assert(validateChainNameFormat(chaincode), "chain_code", tpa->chain_code, "Invalid chain code format",
                                ErrorInvalidFioNameFormat);
                 fio_400_assert(validatePubAddressFormat(tpa->public_address), "public_address", tpa->public_address,
                                "Invalid public address format",
                                ErrorChainAddressEmpty);
 
-                for( auto it = fioname_iter->addresses.begin(); it != fioname_iter->addresses.end(); ++it ) {
-                    if( (it->token_code == token) && (it->chain_code == chaincode)  ){
-                        namesbyname.modify(fioname_iter, actor, [&](struct fioname &a) {
-                            a.addresses[it-fioname_iter->addresses.begin()].public_address = tpa->public_address;
-                        });
-                        wasFound = true;
-                        break;
+                auto it = std::find_if(fioname_iter->addresses.begin(), fioname_iter->addresses.end(),
+                                       find_token(token));
+                if ((it->token_code == token) && (it->chain_code == chaincode)) {
+                    namesbyname.modify(fioname_iter, actor, [&](struct fioname &a) {
+                        a.addresses[it - fioname_iter->addresses.begin()].public_address = tpa->public_address;
+                    });
+                    wasFound = true;
+                } else if (it->token_code == token && it->chain_code != chaincode) {
+                    for (auto it = fioname_iter->addresses.begin(); it != fioname_iter->addresses.end(); ++it) {
+                        if ((it->token_code == token) && (it->chain_code == chaincode)) {
+                            namesbyname.modify(fioname_iter, actor, [&](struct fioname &a) {
+                                a.addresses[it - fioname_iter->addresses.begin()].public_address = tpa->public_address;
+                            });
+                            wasFound = true;
+                            break;
+                        }
                     }
                 }
+
                 if(!wasFound){
-                    fio_400_assert(fioname_iter->addresses.size() != 100, "token_code", tpa->token_code, "Maximum token codes mapped to single FIO Address reached. Only 100 can be mapped.",
-                                   ErrorInvalidFioNameFormat);
+                    fio_400_assert(fioname_iter->addresses.size() != MAX_SET_ADDRESSES, "token_code", tpa->token_code, "Maximum token codes mapped to single FIO Address reached. Only 200 can be mapped.",
```

Comment:
Can we use MAX_SET_ADDRESSES in the error message itself?

Reply:
No, not possible with the current implementation. Templated functions are not available on the contract side.